### PR TITLE
feat: add use_trait_observe and use_dark_effective

### PR DESCRIPTION
**Pull request by maartenbreddels**
Makes it easy to observe traits, and the example application of this
is use_dark_effective.

### CODE REVIEW

**Comment by maartenbreddels** [comment]
* **#551** <a href="https://app.graphite.dev/github/pr/widgetti/solara/551?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maartenbreddels and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: solara/lab/components/theming.py [hunk 1]
```diff
@@ -11,6 +11,21 @@
...
+    """Return True if the frontend is using a dark theme.
+
+    Equivalent of
+
+    ```python
+    solara.use_trait_observe(solara.lab.theme, "dark_effective")
+    ```
+
+    See [use_trait_observe](/api/use_trait_observe).
+
```
Comment (iisakkirotko) [inline solara/lab/components/theming.py:24]:
```suggestion
```

File: solara/lab/components/theming.py [hunk 2]
```diff
@@ -11,6 +11,21 @@
...
+    """
+    return solara.use_trait_observe(solara.lab.theme, "dark_effective")
```
Comment (iisakkirotko) [inline solara/lab/components/theming.py:26]: Why use `solara.lab.theme` here, instead of just `theme`?

**Review by iisakkirotko — CHANGES_REQUESTED** [review]

**Review by iisakkirotko — APPROVED** [review]

**Comment by iisakkirotko** [comment]
Thank you!

**Pull request merged**